### PR TITLE
Add an `--update` CLI arg to `darwin-switch`

### PR DESCRIPTION
**Pull request by yvan-sraka**
This mimics the behaviour of [`nixos-rebuild switch --upgrade`](https://nixos.org/manual/nixos/stable/index.html#sec-upgrading), running `nix-channel --update darwin; nixos-rebuild switch`.

### CODE REVIEW

File: pkgs/nix-tools/darwin-rebuild.sh [hunk 1]
```diff
@@ -112,6 +116,10 @@ done
 
 if [ -z "$action" ]; then showSyntax; fi
 
+if [ "$updateChannels" = true ] ; then
+  nix-channel --update darwin
```
Comment (lockejan) [inline pkgs/nix-tools/darwin-rebuild.sh:120]: Hmm, that would fail in a lot of cases, where people use nix as a deamon service and therefore root owns it.

**Comment by lockejan** [comment]
Can you provide some tests to verify that this works as intended?

File: pkgs/nix-tools/darwin-rebuild.sh (hunk 1 above)
Comment (yvan-sraka) [reply pkgs/nix-tools/darwin-rebuild.sh:120]: You're right… is it reasonable to try to catch all that cases?

**Comment by yvan-sraka** [comment]
> Can you provide some tests to verify that this works as intended?

Yes! Indeed.

File: pkgs/nix-tools/darwin-rebuild.sh (hunk 1 above)
Comment (lockejan) [reply pkgs/nix-tools/darwin-rebuild.sh:120]: You could test if the nix-channel exists and if nix runs as a daemon.
However the latter needs sudo, to inspect the running services.

(There's also the possibility that people have nix-darwin as a user and a root channel, without knowing it. I guess that's currently a bug in the installer. At least to my understanding.)

Comment (dstcruz) [reply pkgs/nix-tools/darwin-rebuild.sh:120]: Oh?! I think I have the scenario described here: root channel and nixpkgs, user nix-darwin. If that's "wrong", how can I clean that up?

Comment (yvan-sraka) [reply pkgs/nix-tools/darwin-rebuild.sh:120]: Maybe that command line argument is a good occasion to throw a user warning/advice if it detects a "wrong" configuration?

**Comment by yvan-sraka** [comment]
Supersede by https://github.com/LnL7/nix-darwin/pull/1014

**Pull request closed without merging**